pystac/catalog.py

```diff
@@ -753,7 +773,17 @@ def save(self, catalog_type: Optional[CatalogType] = None) -> None:
             if root_link and root_link.get_absolute_href() == self.get_self_href():
                 include_self_link = True
 
-        self.save_object(include_self_link=include_self_link)
+        catalog_dest_href = None
+        if dest_href is not None and root.catalog_type in [
+            CatalogType.ABSOLUTE_PUBLISHED
+        ]:
+            rel_href = make_relative_href(self.get_self_href(), self.get_self_href())
```

Comment:
Same as above; we should check for the case where either (or both) of the `get_self_href` calls return `None` and raise a helpful exception here.